src/v/pandaproxy/schema_registry/test/test_json_schema.cc

```diff
@@ -255,6 +255,12 @@ static constexpr auto compatibility_test_cases = std::to_array<
     = R"({"type": "object", "properties": {"a": {"type": "integer", "default": 10}}, "required": ["a"]})",
     .reader_is_compatible_with_writer = false,
   },
+  // array checks: size increase is not allowed
+  {
+    .reader_schema = R"({"type": "array", "minItems": 2, "maxItems": 10})",
+    .writer_schema = R"({"type": "array", "maxItems": 11})",
+    .reader_is_compatible_with_writer = false,
+  },
```

Comment:
nit: is it worth splitting this into 2 tests, one that changes `minItems` and another that changes `maxItems` to show that neither change is allowed (independently of one another)?

Reply:
good point, i'll follow up